libs/ssl-config/src/main/java/org/elasticsearch/common/ssl/PemUtils.java

```diff
@@ -73,6 +72,29 @@ private PemUtils() {
         throw new IllegalStateException("Utility class should not be instantiated");
     }
 
+    /**
+     * Creates a {@link PrivateKey} from the contents of a file and handles any exceptions
+     *
+     * @param path           the path for the key file
+     * @param passwordSupplier A password supplier for the potentially encrypted (password protected) key
+     * @return a private key from the contents of the file
+     */
+    public static PrivateKey readPrivateKey(Path path, Supplier<char[]> passwordSupplier) throws IOException, GeneralSecurityException {
```

Comment:
This is just a new wrapper method that does the special exception handling we need (so that admins get useful error messages).